tests/core/celery/tasks/test_generate_inventory_and_hold_reports.py

```diff
@@ -0,0 +1,279 @@
+    publisher = "My Publisher"
+    checkouts_left = 10
+    terms_concurrency = 5
+    edition = db.edition(data_source_name=ds.name)
+    edition.language = language
+    edition.publisher = publisher
+    edition.title = title
+    edition.medium = edition.BOOK_MEDIUM
+    edition.author = author
+    work = db.work(
+        language="eng",
+        fiction=True,
+        with_license_pool=False,
+        data_source_name=ds.name,
+        presentation_edition=edition,
+        collection=collection,
+        genre="genre_z",
+    )
+
+    genre, ignore = Genre.lookup(db.session, "genre_a", autocreate=True)
+    work.genres.append(genre)
+    work.audience = "young adult"
+
+    licensepool = db.licensepool(
+        edition=edition,
+        open_access=False,
+        data_source_name=ds.name,
+        set_edition_as_presentation=True,
+        collection=collection,
+    )
+
+    days_remaining = 10
+    expiration = utc_now() + timedelta(days=days_remaining)
+    db.license(
+        pool=licensepool,
+        status=LicenseStatus.available,
+        checkouts_left=checkouts_left,
+        terms_concurrency=terms_concurrency,
+        expires=expiration,
+    )
+    db.license(
+        pool=licensepool,
+        status=LicenseStatus.unavailable,
+        checkouts_left=1,
+        terms_concurrency=1,
+        expires=utc_now(),
+    )
+
+    patron1 = db.patron(library=library)
+    patron2 = db.patron(library=library)
+    patron3 = db.patron(library=library)
+    patron4 = db.patron(library=library)
+
+    # this one should be counted because the end is in the future.
+    hold1, _ = get_one_or_create(
+        db.session,
+        Hold,
+        patron=patron1,
+        license_pool=licensepool,
+        position=1,
+        start=utc_now(),
+        end=utc_now() + timedelta(days=1),
+    )
+
+    # this one should be counted because the end is None
+    hold2, _ = get_one_or_create(
+        db.session,
+        Hold,
+        patron=patron2,
+        license_pool=licensepool,
+        start=utc_now(),
+        end=None,
+    )
+
+    # this hold should be counted b/c the position is > 0
+    hold3, _ = get_one_or_create(
+        db.session,
+        Hold,
+        patron=patron3,
+        license_pool=licensepool,
+        start=utc_now() - timedelta(days=1),
+        end=utc_now() - timedelta(minutes=1),
+        position=1,
+    )
+
+    # this hold should not be counted because the end is neither in the future nor unset and the position is zero
+    hold4, _ = get_one_or_create(
+        db.session,
+        Hold,
+        patron=patron4,
+        license_pool=licensepool,
+        start=utc_now(),
+        end=utc_now() - timedelta(minutes=1),
+        position=0,
+    )
+
+    shared_patrons_in_hold_queue = 4
+    licensepool.patrons_in_hold_queue = shared_patrons_in_hold_queue
+
+    assert library.id
+
+    job = GenerateInventoryAndHoldsReportsJob(
+        mock_session_maker, library.id, email_address=email
```

Comment:
If you took `delete_attachments` and `send_email` as args here, this could be re-written as:
```python
    send_email_mock = create_autospec(job.services.email.container.send_email)
    # for testing, don't delete the files associated with the attachments so we can read them after the script
    # runs
    job = GenerateInventoryAndHoldsReportsJob(
        mock_session_maker, library.id, email_address=email, send_email=send_email_mock, delete_attachments=False
    )
```